Enable ptrace on all Linux platforms

geofft opened the pull request:
Nothing that nix currently binds is architecture-specific, and Android
supports ptrace just as much as non-Android Linux.

geofft commented:
I noticed this in the course of adding a test for #566.... but also in the course of adding a test for #566 I learned that qemu-user doesn't emulate ptrace on anything other than x86 or x86-64. So getting test coverage for this change is going to be hard.

Susurrus commented:
As for tests, couldn't you have a test that just makes any call and even if an `Err` is returned, it passes? This way we could at least confirm that ptrace is available on the correct platforms.

Susurrus commented:
Additionally we can run actual tests only on non-qemu targets.

Susurrus commented:
@geofft This needs a CHANGELOG entry and a rebase and it should be good to merge.

geofft commented:
@Susurrus rebased, added a changelog entry, and added a test that ptracing yourself fails with either EPERM (expected) or ENOSYS (qemu-user). Lemme know if you need anything else and sorry for being slow to do this!

Susurrus commented:
LGTM. Eventually I'd like to have a few `ptrace` tests that actuall do stuff if possible, but we'll defer on that for now. This test you added is at least good enough for this PR. Thanks for adding it!

bors r+
